airbyte-server/src/test/java/io/airbyte/server/migration/RunMigrationTest.java

```diff
@@ -137,6 +141,8 @@ private void assertSourceDefinitions(ConfigRepository configRepository) throws J
         .stream()
         .collect(Collectors.toMap(c -> c.getSourceDefinitionId().toString(), c -> c));
     assertTrue(sourceDefinitions.size() >= 59);
+    // the definition is not present in latest seeds so it should be deleted
+    assertFalse(sourceDefinitions.containsKey("d2147be5-fa36-4936-977e-f031affa5895"));
```

Comment:
is there a bit that also shows old definition with a usage will not be deleted?

Reply:
The following assertion makes sure that the source that is being used stays on the old version and doesnt get upgraded
```
final StandardSourceDefinition mysqlDefinition = sourceDefinitions.get("435bb9a5-7887-4809-aa58-28c27df0d7ad");
    assertEquals("0.2.0", mysqlDefinition.getDockerImageTag());
    assertEquals("MySQL", mysqlDefinition.getName());
```
But there is no assertion for the scenario where a connector existed in the past and is being used and should not get deleted

Reply:
I can add one, should not be too tough